packages/desktop-client/src/components/common/ToggleMenu.tsx

```diff
@@ -0,0 +1,203 @@
+import { type ReactNode, useRef, useState, useEffect } from 'react';
+
+import { type CSSProperties, theme } from '../../style';
+
+import { Text } from './Text';
+import { Toggle } from './Toggle';
+import { View } from './View';
+
+type KeybindingProps = {
+  keyName: ReactNode;
+};
+
+function Keybinding({ keyName }: KeybindingProps) {
+  return (
+    <Text style={{ fontSize: 10, color: theme.menuKeybindingText }}>
+      {keyName}
+    </Text>
+  );
+}
+
+type MenuItem = {
+  type?: string | symbol;
+  name: string;
+  disabled?: boolean;
+  toggle?: boolean;
+  tooltip?: string;
+  text: string;
+  key?: string;
+  isOn?: boolean;
+  style?: CSSProperties;
+};
+
+type ToggleMenuProps = {
+  header?: ReactNode;
+  footer?: ReactNode;
+  items: Array<MenuItem | typeof ToggleMenu.line>;
+  onMenuSelect: (itemName: MenuItem['name']) => void;
+  style?: CSSProperties;
+};
+
+export function ToggleMenu({
+  header,
+  footer,
+  items: allItems,
+  onMenuSelect,
+  style,
+}: ToggleMenuProps) {
+  const elRef = useRef(null);
+  const items = allItems.filter(x => x);
+  const [hoveredIndex, setHoveredIndex] = useState(null);
+
+  useEffect(() => {
+    const el = elRef.current;
+    el.focus();
+
+    const onKeyDown = e => {
```

Comment:
Is there a reason why we are using a ref to set the event listener? Can we put this handler on the `View` itself via `onKeyUp` or `onKeyDown` props?